src/passes/OptimizeInstructions.cpp

```diff
@@ -64,16 +64,57 @@ Index getMaxBits(Expression* curr, LocalInfoProvider* localInfoProvider) {
   } else if (auto* binary = curr->dynCast<Binary>()) {
     switch (binary->op) {
       // 32-bit
+      case RotLInt32:
+      case RotRInt32:
       case AddInt32:
       case SubInt32:
       case MulInt32:
-      case DivSInt32:
-      case DivUInt32:
-      case RemSInt32:
-      case RemUInt32:
-      case RotLInt32:
-      case RotRInt32:
+      case RemSInt32: {
+        if (auto* const_ = binary->right->dynCast<Const>()) {
```

Comment:
we usually do `auto* c = ` for this. please do that in new code, and we should fix the old code too eventually.